linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/resultset/StorageResultSetReader.java

```diff
@@ -0,0 +1,179 @@
+    this(resultSet, new ByteArrayInputStream(value.getBytes(Dolphin.CHAR_SET)));
+  }
+
+  public void init() throws IOException {
+    String resType = Dolphin.getType(inputStream);
+    if (!StringUtils.equals(resultSet.resultSetType(), resType)) {
+      throw new RuntimeException(
+          "File type does not match(文件类型不匹配): "
+              + ResultSetFactory.resultSetType.getOrDefault(resType, "TABLE"));
+    }
+  }
+
+  public byte[] readLine() {
+    int rowLen = 0;
+    try {
+      rowLen = Dolphin.readInt(inputStream);
+    } catch (StorageWarnException | IOException e) {
+      logger.info("Read finished(读取完毕)");
+      return null;
+    }
+
+    byte[] rowBuffer = new byte[0];
+    int len = 0;
+
+    while (rowLen > 0 && len >= 0) {
+      if (rowLen > READ_CACHE) {
+        len = StorageUtils.readBytes(inputStream, bytes, READ_CACHE);
+      } else {
+        len = StorageUtils.readBytes(inputStream, bytes, rowLen);
+      }
+
+      if (len > 0) {
+        rowLen -= len;
+        rowBuffer = Arrays.copyOf(rowBuffer, rowBuffer.length + len);
+        System.arraycopy(bytes, 0, rowBuffer, rowBuffer.length - len, len);
+      }
+    }
+    rowCount++;
+    return rowBuffer;
+  }
+
+  @Override
+  public Record getRecord() {
+    if (metaData == null) throw new RuntimeException("Must read metadata first(必须先读取metadata)");
+    if (row == null) {
+      throw new RuntimeException(
+          "Can't get the value of the field, maybe the IO stream has been read or has been closed!(拿不到字段的值，也许IO流已读取完毕或已被关闭！)");
+    }
+    return row;
+  }
+
+  public void setFs(Fs fs) {
+    this.fs = fs;
+  }
+
+  public Fs getFs() {
+    return fs;
+  }
+
+  @Override
+  public MetaData getMetaData() {
+    if (metaData == null) {
+      try {
+        init();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    metaData = deserializer.createMetaData(readLine());
+    return metaData;
+  }
+
+  @Override
+  public int skip(int recordNum) throws IOException {
+    if (recordNum < 0) return -1;
+
+    if (metaData == null) getMetaData();
+    for (int i = recordNum; i > 0; i--) {
+      try {
+        inputStream.skip(Dolphin.readInt(inputStream));
+      } catch (Throwable t) {
+        return recordNum - i;
+      }
+    }
+    return recordNum;
+  }
+
+  @Override
+  public long getPosition() throws IOException {
+    return rowCount;
+  }
+
+  @Override
+  public boolean hasNext() throws IOException {
+    if (metaData == null) getMetaData();
+    byte[] line = readLine();
+    if (line == null) return false;
+    row = deserializer.createRecord(line);
+    if (row == null) return false;
+    return true;
+  }
+
+  @Override
+  public long available() throws IOException {
+    return inputStream.available();
+  }
+
+  @Override
+  public void close() throws IOException {
+    inputStream.close();
+    if (this.fs != null) {
+      this.fs.close();
```

Comment:
Should use IOUtils.closeQuietly

Reply:
Thanks for your time ,I already used IOUtils.closeQuietly.